### PR TITLE
Adding netket

### PR DESCRIPTION
**Pull request by PhilipVinc**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by PhilipVinc** [comment]
@conda-forge-admin, please lint

**Comment by PhilipVinc** [comment]
cc @gcarleo
If you wish to be listed as a recipe maintainer please write it below.

**Comment by PhilipVinc** [comment]
@conda-forge-admin, please restart ci

**Comment by gcarleo** [comment]
Yes I would like to be a maintaner of this, thanks

**Comment by PhilipVinc** [comment]
@conda-forge-admin, please restart ci

**Comment by PhilipVinc** [comment]
@conda-forge-admin, please restart ci

**Comment by PhilipVinc** [comment]
@conda-forge-admin, please restart ci

**Comment by PhilipVinc** [comment]
@conda-forge-admin, please restart ci

**Comment by PhilipVinc** [comment]
@conda-forge-admin, please restart ci

**Comment by PhilipVinc** [comment]
@conda-forge-admin, please ping team

**Comment by PhilipVinc** [comment]
There are a [bunch of errors on osx](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=147786&view=logs&j=c572e5d5-7ba7-55ce-5575-7ce1d9f7161e&t=81967b5f-0283-5c0b-87c1-e8fdeef842a3&l=865) 
```
Fixing linking of @rpath/libmpi.12.dylib in /usr/local/miniconda/conda-bld/netket_1587575547023/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.7/site-packages/netket/_C_netket.cpython-37m-darwin.so
New link location is None
Fixing linking of @rpath/libomp.dylib in /usr/local/miniconda/conda-bld/netket_1587575547023/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.7/site-packages/netket/_C_netket.cpython-37m-darwin.so
New link location is None
Fixing linking of /System/Library/Frameworks/Accelerate.framework/Versions/A/Accelerate in /usr/local/miniconda/conda-bld/netket_1587575547023/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.7/site-packages/netket/_C_netket.cpython-37m-darwin.so
New link location is None
Fixing linking of @rpath/libc++.1.dylib in /usr/local/miniconda/conda-bld/netket_1587575547023/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.7/site-packages/netket/_C_netket.cpython-37m-darwin.so
New link location is None
Fixing linking of /usr/lib/libSystem.B.dylib in /usr/local/miniconda/conda-bld/netket_1587575547023/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.7/site-packages/netket/_C_netket.cpython-37m-darwin.so
New link location is None

  ERROR :: lib/libc++.1.dylib not in prefix_owners
  ERROR :: lib/libmpi.12.dylib not in prefix_owners
  ERROR :: lib/libomp.dylib not in prefix_owners
```

which I don't know why they arise nor how to solve them. However, the library works fine and tests pass so apparently they don't impact the resulting binary.

Also I don't know how to properly address those two warnings: In conda-forge I saw the instructions to link to libblas. Moreover, the package requires python so... what should I do? 
```
WARNING (netket): run-exports library package conda-forge::libblas-3.8.0-8_h1ddc27c_netlib in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (netket): run-exports library package conda-forge::python-3.7.6-h90870a6_5_cpython in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

File: recipes/netket/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - numba >=0.48
+    - mpich
+    - mpi4py
+    - tqdm
+    - libblas
+    - libcblas
+    - libcxx
+    - libgomp  # [linux]
+    - openmp  # [linux]
+    - llvm-openmp  # [osx]
```
Comment (isuruf) [inline recipes/netket/meta.yaml:46]: you can remove the 6 lines above

File: recipes/netket/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - cmake >=3.12
+    - ninja
+  host:
+    - pip
+    - python =3.7
+    - mpich
+    - libblas
```
Comment (isuruf) [inline recipes/netket/meta.yaml:27]: remove this line

File: recipes/netket/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - cmake >=3.12
+    - ninja
+  host:
+    - pip
+    - python =3.7
+    - mpich
+    - libcblas
+    - openmp  # [linux]
```
Comment (isuruf) [inline recipes/netket/meta.yaml:28]: Remove this line too

File: recipes/netket/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    - libgomp  # [linux]
+    - llvm-openmp  # [osx]
+    - libcxx
```
Comment (isuruf) [inline recipes/netket/meta.yaml:31]: Same here

File: recipes/netket/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - cmake >=3.12
+    - ninja
+  host:
+    - pip
+    - python =3.7
```
Comment (isuruf) [inline recipes/netket/meta.yaml:25]: See https://conda-forge.org/docs/maintainer/adding_pkgs.html#skipping-builds on how to skip python versions

File: recipes/netket/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
+{% set name = "netket" %}
+{% set version = "2.1" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://github.com/netket/netket/archive/v2.1.tar.gz
```
Comment (isuruf) [inline recipes/netket/meta.yaml:9]: Use jinja for `2.1`

File: recipes/netket/meta.yaml (hunk 1 above)
Comment (PhilipVinc) [reply recipes/netket/meta.yaml]: the 6 lines above are 
```
     - libblas		
     - libcblas		
     - libcxx		
     - libgomp  # [linux]		
     - openmp  # [linux]		
     - llvm-openmp  # [osx]		
```
maybe I should keep libcblas? Having removed all of them causes a build failure on linux

Comment (PhilipVinc) [reply recipes/netket/meta.yaml]: see https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=147811&view=logs&j=10bda42a-d420-5c28-a43f-bc4e66405e5c&t=bef775d0-a1a2-55a3-9721-17147029baf0&l=1357

File: recipes/netket/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - cmake >=3.12
+    - ninja
+  host:
+    - pip
+    - python
+    - mpich
+    - libcblas
```
Comment (isuruf) [inline recipes/netket/meta.yaml:27]:
```suggestion
    - libblas
```

**Comment by PhilipVinc** [comment]
Thanks @isuruf .
I implemented all the above. Build passes. There are still those weird errors on Mac, though...

**Comment by PhilipVinc** [comment]
@isuruf should I do something else or this is good to be merged?

**Comment by PhilipVinc** [comment]
@conda-forge-admin, please ping team 
This is real for merging

**Pull request merged**